src/core/LinkedModelGroups.cpp

```diff
@@ -0,0 +1,359 @@
+/*
+ * LinkedModelGroups.cpp - base classes for groups of linkable models
+ *
+ * Copyright (c) 2019-2019 Johannes Lorenz <j.git$$$lorenz-ho.me, $$$=@>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "LinkedModelGroups.h"
+
+#include <QDomDocument>
+#include <QDomElement>
+
+#include "AutomatableModel.h"
+
+
+
+
+/*
+	LinkedModelGroup
+*/
+
+
+
+
+void LinkedModelGroup::makeLinkingProc()
+{
+	for (std::size_t i = 0; i < m_models.size(); ++i)
+	{
+		BoolModel* bmo = new BoolModel(true, this, tr("Link channels"));
+		m_linkEnabled.push_back(bmo);
+		connect(bmo, SIGNAL(dataChanged()), this, SLOT(linkStateChangedSlot()));
```

Comment:
This is really cool, it would save me 10 lines! Your example looks right, but it does not compile:

```
LinkedModelGroups.cpp:48:3: error: no matching member function for call to 'connect'
...
qobject.h:308:13: note: candidate template ignored: substitution failure [with Func1 = const char *, Func2 = (lambda at /home/johannes/cprogs/lmms/master/src/core/LinkedModelGroups.cpp:49:5)]: no type named
      'Object' in 'QtPrivate::FunctionPointer<const char *>'
```

Do you have any idea what I might be doing wrong? Would you mind checking on your desktop?

Reply:
I've just tried it and it compiles for me - are you sure you've changed the second argument from `SIGNAL(dataChanged())` to `&BoolModel::dataChanged`?

Reply:
Oh, I didn't do that. That was the issue, thanks.